Use requirejs to bundle/minify extensions, copy only what we need to dist/, part of #634

humphd opened the pull request:
I've reworked our extension loading to properly package extensions in `dist/` as minified requirejs bundles, and changed how we copy files over from `src/` to `dist/` so we only have what we need.  I'll need this for my next trick with service worker cache tuning in #634.

Pomax commented on README.md:
```diff
@@ -67,53 +67,11 @@ should host Bramble's iframe, see `src/hosted.js`.
-extension lists.
+# Extension Loading
+
+Bramble loads a set of extensions defined in `src/extensions/bramble-extensions.json`. You can
```
small nit, but worth `[]()` link wrapping to jump to that file on github?

humphd replied:
Good suggestion. I think we don't do this anywhere in this file, and should.

humphd commented:
OK, I've rebased this, and updated to deal with changes to the extensions we're loading.  This can get reviewed, I think it's basically good to go, and a huge win overall.

humphd commented:
@gideonthomas I fixed the missing files we saw the other day when I showed you this.  I've also decided to remove one of the extensions I added, and filed a bug here to maybe include it later, https://github.com/mozilla/brackets/issues/681.

I think this is ready to go now.  Test with `npm run build` and then run from `dist/hosted.html` and make sure you only see that 1 404 for require.

After this lands, I'll do another PR to fix how the SW cache list works (e.g., just use `dist/**/*`)

humphd commented:
One more commit to not copy `styles/fonts/**/*`, which is the Source Code Pro fonts we don't use.

humphd commented:
I noticed a few other things we could improve, and filed:

* Minify CSS - https://github.com/mozilla/brackets/issues/683
* Require/Minify JavaScriptCodeHints worker stuff - https://github.com/mozilla/brackets/issues/684
